coreneuron/apps/main1.cpp

```diff
@@ -126,8 +126,8 @@ char* prepare_args(int& argc, char**& argv, int use_mpi, const char* arg) {
     return first;
 }
 
-int corenrn_embedded_run(int nthread, int have_gaps, int use_mpi, int use_fast_imem, const char* arg) {
-    corenrn_embedded = 1;
+int corenrn_embedded_run(int nthread, bool have_gaps, int use_mpi, int use_fast_imem, const char* arg) {
```

Comment:
I wouldn't change this because of [use here in NEURON](https://github.com/neuronsimulator/nrn/blob/ca2d8a36d00494423c2feb2794d6a0cd184f308f/src/nrniv/nrnbbcore_write.cpp#L1957).

Reply:
Thanks for seeing that, I'm really afraid of breaking things when modifying CoreNeuron.

A new reason to better define the public API.

Reply:
Agree! let's do that as we start refactoring!